File: src/pkg/runtime/slice.c
// Copyright 2014 The Go Authors. All rights reserved.
// Use of this source code is governed by a BSD-style
// license that can be found in the LICENSE file.

#include "runtime.h"
#include "arch_GOARCH.h"

void
runtime·printslice_m(G *gp)
{
	void *array;
	uintptr len, cap;

	array = g->m->ptrarg[0];
	g->m->ptrarg[0] = nil;
	len = g->m->scalararg[0];
	cap = g->m->scalararg[1];

	runtime·prints("[");
	runtime·printint(len);
	runtime·prints("/");
	runtime·printint(cap);
	runtime·prints("]");
	runtime·printpointer(array);

	runtime·gogo(&gp->sched);
}

File: src/cmd/compile/internal/types2/slice.go
// Copyright 2011 The Go Authors. All rights reserved.
// Use of this source code is governed by a BSD-style
// license that can be found in the LICENSE file.

package types2

// A Slice represents a slice type.
type Slice struct {
	elem Type
}

// NewSlice returns a new slice type for the given element type.
func NewSlice(elem Type) *Slice { return &Slice{elem: elem} }

// Elem returns the element type of slice s.
func (s *Slice) Elem() Type { return s.elem }

func (s *Slice) Underlying() Type { return s }
func (s *Slice) String() string   { return TypeString(s, nil) }
